File: src/repositories/subscription.repository.ts
import type {Collection, Filter} from "mongodb";
import {getCoreDatabase} from "@/config/database";
import type {Subscription} from "@/types/subscription/subscription";
import {getSubscriptionCollection} from "@/repositories/collections/core.collections";

export class SubscriptionRepository {

  constructor() {
  }

  private getCollection(): Collection<Subscription> {
    return getSubscriptionCollection();
  }

  async create(subscription: Omit<Subscription, "_id">): Promise<Subscription> {
    const result = await this.getCollection().insertOne(subscription as any);
    return {...subscription, _id: result.insertedId} as Subscription;
  }

  async delete(companyId: string): Promise<boolean> {
    const result = await this.getCollection().deleteOne({companyId: companyId} as Filter<Subscription>);

    return result.deletedCount > 0;
  }

  async findByCompanyId(companyId: string): Promise<Subscription | null> {
    const doc = await this.getCollection().findOne({companyId: companyId} as Filter<Subscription>);

    return doc as Subscription | null;
  }

}

File: fisoft/src/main/kotlin/repository/SubscriptionRepository.kt
package com.ilyasdemirkiran.repository

import com.ilyasdemirkiran.config.Database
import com.ilyasdemirkiran.types.subscription.Subscription
import com.mongodb.client.model.Filters
import com.mongodb.kotlin.client.coroutine.MongoCollection
import kotlinx.coroutines.flow.firstOrNull
import org.bson.types.ObjectId

class SubscriptionRepository {
    private fun getCollection(): MongoCollection<Subscription> {
        return Database.getCoreDatabase().getCollection<Subscription>("subscriptions")
    }

    suspend fun create(sub: Subscription): Subscription {
        val id = sub._id ?: ObjectId().toHexString()
        val created = sub.copy(_id = id)
        getCollection().insertOne(created)
        return created
    }

    suspend fun findByCompanyId(companyId: String): Subscription? {
        return getCollection().find(Filters.eq("companyId", companyId)).firstOrNull()
    }

    suspend fun delete(companyId: String): Boolean {
        val res = getCollection().deleteOne(Filters.eq("companyId", companyId))
        return res.deletedCount > 0
    }
}
